drivers/adc/ad4170/ad4170.c

```diff
@@ -0,0 +1,2119 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, INTELLECTUAL PROPERTY RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*******************************************************************************/
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdbool.h>
+#include <string.h>
+#include "no_os_error.h"
+#include "no_os_delay.h"
+#include "no_os_crc8.h"
+#include "ad4170.h"
+
+NO_OS_DECLARE_CRC8_TABLE(ad4170_crc8);
+
+struct ad4170_config ad4170_config_reset = {
+	.pin_muxing = {
+		.sync_ctrl = AD4170_SYNC_STANDARD,
+	},
+	.channel_en = 0x1,
+	.map = {
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1},
+		{.ainm = AD4170_AIN1}
+	},
+	.setups = {
+		{ /* setup 0 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 1 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 2 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 3 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 4 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 5 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 6 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+		{ /* setup 7 */
+			.afe = {
+				.bipolar = true,
+				.ref_select = AD4170_REFIN_REFOUT,
+			},
+			.filter_fs = 0x4,
+			.gain = 0x555555,
+		},
+	},
+	.ref_control = {
+		.ref_en = true,
+	},
+	.fir_control = {
+		.fir_length = 1,
+	}
+};
+
+static inline bool _is_addr_valid(struct ad4170_dev *dev, uint32_t reg_addr)
+{
+	uint8_t reg_size = AD4170_TRANSF_LEN(reg_addr);
+	uint32_t addr = AD4170_ADDR(reg_addr);
+
+	/* max readable address in short instruction mode */
+	if (dev->spi_settings.short_instruction && addr > 0x3F)
```

Comment:
Should we use macros for values like this one?

Reply:
i'd leave it as-is because it has something to do with a certain spi standard and we don't have headers for that